tests/irdl/test_operation_definition.py

```diff
@@ -627,3 +631,41 @@ def test_multiple_inheritance_op():
     )
     with pytest.raises(DiagnosticException):
         op_attr_fail.verify()
+
+
+@irdl_op_definition
+class EntryArgsOp(IRDLOperation):
+    name = "test.entry_args"
+    body = opt_region_def(entry_args=RangeOf(EqAttrConstraint(i32)))
+
+    traits = frozenset((NoTerminator(),))
+
+
+def test_entry_args_op():
+    op = EntryArgsOp.create()
+    op.verify()
+
+    op = EntryArgsOp.create(regions=[Region(Block(arg_types=[]))])
+    op.verify()
+    op = EntryArgsOp.create(regions=[Region(Block(arg_types=[i32]))])
+    op.verify()
+    op = EntryArgsOp.create(regions=[Region(Block(arg_types=[i32, i32]))])
+    op.verify()
+
+    op = EntryArgsOp.create(regions=[Region(Block(arg_types=[i64]))])
+    with pytest.raises(
+        DiagnosticException,
+        match="""\
+Operation does not verify: region #0 entry arguments do not verify:
+Expected attribute i32 but got i64""",
+    ):
```

Comment:
SHouldnt this be less hard coded?

Reply:
What do you mean?

Reply:
I like testing explicitely the expressed error, because an easy regression in that area is, if you just test that verification fails, for some reasons verification might fail for an unrelated reason. Then, nothing is checking the exact tested verification feature anymore!